### ISSUE TITLE
API is returning all nodes regardless of the publishing status

### ISSUE THREAD

**Issue by dmedeiros32** [Southbank-Centre/OpenEvent#78]
Hi Jorge and Tassos, I've got another issue - the api seems to ignore the published status of the events and other nodes - in my setup it returns all items regardless of the publishing status. This is critical for our publishing environment, as new events may be created and published only after review.

Ideally, the status fields would be included, such as Published Status, Send to Home Page, and Sticky.

Regards,
Danilo

**Comment by tassoskoutlas** [comment]
Hey @dmedeiros32,
can you check in anonymous session if unpublished content is returned? It should take into account the publish status and user permissions for that matter.

Cheers.

**Comment by dmedeiros32** [comment]
Dear Tassos, I have confirmed that even non-published events are being displayed. Also, is there a way to filter results to show only items that are pushed to the home page? Any chance to get them ordered by a field? If those features are not implemented, would you kindly point to the best way to implement them so we can do it? 

thank you once again

**Comment by dmedeiros32** [comment]
Tassos, did you have the time to look at the issue by any chance? Thank you once again

**Comment by samueljmurray** [comment]
Hi @dmedeiros32 

**Unpublished content displaying**

I've just had a look at your issue with unpublished content and I'm not able to replicate.

I installed a fresh Drupal site with the OpenEvent profile and logged in as the root user. I then created an Event node, and kept it unpublished. Accessing the content in an **incognito** browser window (so there is no authenticated Drupal session), I get the following:

![image](https://cloud.githubusercontent.com/assets/4062874/10547230/08096d00-742b-11e5-99ee-c5effa7ef459.png)

Using the basic RestWS endpoint for the node, I get this:

![image](https://cloud.githubusercontent.com/assets/4062874/10547262/3958db02-742b-11e5-9244-726bfed31c5c.png)

Are you using one of the core content types that comes with OpenEvent (Event, Organization, Person, Place), or one that you have created yourself?
What is the URL you're using to access the content?
Are you using the latest version of OpenEvent (7.x-0.8) or an earlier one?

**Adding fields to API output**

Try implementing `hook_restws_response_alter`. Refer to line 257 of [oe_schemaorg_rest.module](https://github.com/Southbank-Centre/OpenEvent/blob/master/modules/custom/oe_schemaorg_rest/oe_schemaorg_rest.module) in this repo to see how we've used it.

**Ordering**

There is some ordering possible on the output of the API. Visit http://your-drupal-installation/api/doc and look under the **Sorting** heading for the content type you want to sort.

Hope that helps.

Sam

**Comment by dmedeiros32** [comment]
Thank You Sam. The issue occurs when asking for all events:

http://www.mda.32bits.com.br/api/event.json (shows event id=39)

Please note that the event with id = 39 is not published:

http://www.mda.32bits.com.br/pt-br/api/event/39.json (403 Forbidden)

Thanks a lot for all the help!

**Comment by samueljmurray** [comment]
Hi @dmedeiros32 thanks - yep that's a bug.

I've created an issue on drupal.org because we'll be managing things from there more now. I've also uploaded a patch. Feel free to try it out.

https://www.drupal.org/node/2595237

Sam

**Comment by dmedeiros32** [comment]
Sam, thank you. I'll move to the drupal.org site.
